sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py

```diff
@@ -278,15 +279,17 @@ def _create_tasks_for_claimed_ownership(
                     await self._process_error(partition_context, error)
                     break
         finally:
-            await self._consumers[partition_id].close()
-            del self._consumers[partition_id]
-            await self._close_partition(
-                partition_context,
-                CloseReason.OWNERSHIP_LOST if self._running else CloseReason.SHUTDOWN,
-            )
-            await self._ownership_manager.release_ownership(partition_id)
-            if partition_id in self._tasks:
-                del self._tasks[partition_id]
+            try:
+                await self._consumers[partition_id].close()
+                del self._consumers[partition_id]
+                await self._close_partition(
+                    partition_context,
+                    CloseReason.OWNERSHIP_LOST if self._running else CloseReason.SHUTDOWN,
+                )
```

Comment:
would it be better to put the callback part into `finally`?
suppose the `close` triggers error and then it would go directly to the finally part, user's `on_close` callback is not called.

Reply:
I guess to expand on @yunhaoling 's point - what errors did you observe that led to this change? And where did they occur?
Other lines in this section also might be worth separating out into separate try/finally  - but it depends on what kind of errors we're targeting

Reply:
Potentially release_ownership may raise an exception. Plus, CancelledError may happen anywhere.
consumer.close() itself doesn't raise an exception, _close_partition() doesn't raise an exception.

I pushed some new code to wrap the cleanup of _receive() method into _clean_receive() and asyncio.shield the clean in _receive(). This way we can guarantee the consumer.close() and other methods are run, not interrupted by CancelledError. 
With try... finally, the finished task will be removed from self._tasks no matter what happens.

One try...finally should be enough after making these changes.